pet-store-vanilla-js/app.js

```diff
@@ -34,49 +38,52 @@ function enableAddPetButton() {
   mainPageElements.addPetButton.style.opacity = '1';
 }
 
-async function fetchAndCachePetKinds() {
-  const petKinds = await getPetKinds();
-  if (!petKinds) {
+async function tryFetchPetKinds() {
+  let fetchedPetKinds;
+
+  try {
+    const petKinds = await getPetKinds();
+    for (let kind of petKinds) {
+      petKindsEnum[kind.value] = kind.displayName;
+  
+      const petKindOption = document.createElement('option');
+      petKindOption.innerText = kind.displayName;
+      petKindOption.value = kind.value;
+      formElements.kind.append(petKindOption);
+    }
+    fetchedPetKinds = true;
+  } catch (err) {
+    fetchedPetKinds = false;
+    console.log(err);
```

Comment:
`console.error` not `console.log` and it has to be the first thing you do when catching the error.